### DISCUSSION TITLE
ModuleNotFoundError: No module named 'raven'

### DISCUSSION THREAD

**Discussion by elianalousada** [neuropoly/axondeepseg#153] [Help needed] [answered]
Hello. I'm trying to start training the algorithm and this is the error I get when importing the libraries. Any thoughts? Thank you.

**Comment by mathieuboudreau** [comment]
Hi Eliana,

Try installing the pip requirements again (e.g. in the axondeepseg/ folder, run `pip install -e .`). The module 'raven' is an external Python module that needs to be installed, which we added to the requirements a few months ago.

**Comment by elianalousada** [comment]
Thank you Mathieu, it worked. :)
 Should be different topic, but anyway... Got this error when trying to segment:
ValueError: Cannot feed value of shape (1, 216, 205) for Tensor 'Placeholder:0', which has shape '(?, 512, 512)'
Can't seem to work around it.

**Comment by mathieuboudreau** [comment]
Great - happy that's one thing that's cleared up!

As for your second point, two questions:

1 - What version of AxonDeepSeg are you running (if you downloaded it from GitHub look inside axondeepseg/AxonDeepSeg/_init_.py; if you downloaded it with the `pip` command, run `pip show axondeepseg`).
2 - What command are you running 

Typically, I believe that error occurs because in ADS because images are resampled to a standard unit (0.1 um for SEM, 0.01 um for TEM), and then separated into overlapping patches of 512x512. If the image has less than 512 pixels in either dimension after resampling to that resolution, then that error usually comes up at the patching stage. 

If you're using the command line interface (`axondeepseg`), then that error should not happen if you're using the the version `2.2dev` (but you'll see a more descriptive warning instead), but this version hasn't been released on PyPi yet (hence the `dev` part). If you're encountering this when running the code within python or a notebook (e.g. training a new model), then there isn't a more descriptive exception handling implemented yet and would explain the message that you saw (sorry about that!).

**Comment by mathieuboudreau** [comment, marked as answer]
As a temporary workaround, you could assign a larger pixel size to the image you want to segment (e.g. if using SEM, add the flag `-s 0.1` (default resolution for SEM), then try larger numbers like 0.2, 0.3). This will make ADS upsample (& interpolate) your image, which should then have enough voxels to create at least one patch of size 512x512.

This should only be for testing purpose though, it's not our recommended procedure. You should know the resolution of your histogram images beforehand, and set the resolution for each one. If you sinply have a small image size, you could also try increasing padding it with white voxel, or mirroring it in both directions.

**Comment by elianalousada** [comment]
Indeed if I change the pixel size it works. But not sure about that being the correct way to work around this issue.

**Comment by mathieuboudreau** [comment]
What's the size of your image (# pixel **x** # pixel), what's the resolution of your image, and what's the modality flag that you're giving it (SEM or TEM)?

**Comment by elianalousada** [comment]
Hello Mathieu, I sent you an e-mail. Not sure if you got it. I've been having some more issues, can we skype onde of these days? Thank you in advance.

**Comment by mathieuboudreau** [comment]
@elianalousada Hi Eliana! Sorry, I just noticed it, it got caught in my spam folder for some reason. I will answer you now.
